docs/templating/schema.md

```diff
@@ -149,371 +130,441 @@ spec:
 
 ### Type Restrictions
 
-1. **Map keys must be strings**:
-   ```yaml
-   # Valid
-   labels: "map<string>"
-   ports: "map<integer>"    # Keys are strings, values are integers
-
-   # Invalid - will error
-   data: "map<integer, string>"  # Can't have integer keys
-   ```
-
-2. **No generic object type**:
-   ```yaml
-   # Invalid
-   config: object
-
-   # Valid alternatives
-   config: "map<string>"    # For dynamic keys
-   config:                  # For known structure
-     key1: string
-     key2: integer
-   ```
-
-3. **Custom types must be defined in types section**:
-   ```yaml
-   # Won't work without definition
-   mount: MountConfig
-
-   # Must define first
-   types:
-     MountConfig:
-       path: string
-       readOnly: boolean
-   ```
+- **Map keys must be strings**: `ports: "map<integer>"` (values are integers, keys always strings)
+- **No generic object type**: Use `map<string>` for dynamic keys or define structure explicitly
+- **Custom types must be defined**: Reference only types defined in `schema.types` section
 
 ## Escaping and Special Characters
 
-### Quoting Values
-
-Values containing special characters must be quoted. OpenChoreo supports both single and double quote styles:
+### Quoting and Escaping
 
 ```yaml
-# Single quotes (use '' to escape single quotes)
+# Single quotes: double to escape (use for values with double quotes)
 description: "string | default='User''s timezone'"
-jsonPath: "string | default='.status.conditions[?(@.type==\"Ready\")].status'"
 
-# Double quotes (use backslash escaping)
-pattern: "string | default=\"^[a-z]+\\d{3}$\""
-message: "string | default=\"Value must be \\\"quoted\\\"\""
+# Double quotes: backslash escape (use for regex patterns)
+pattern: "string | default=\"^[a-z]+\\\\d{3}$\""
+
+# Pipes in values must be quoted
+format: 'string | pattern="a|b|c"'
+
+# Enum values with spaces/commas - quote each value
+size: 'string | enum="extra small","small","medium","large"'
+format: 'string | enum="lastname, firstname","firstname lastname"'
 ```
 
-### Pipes in Values
+**Rules:**
+- Single quotes: `''` escapes `'`
+- Double quotes: `\\` escapes `\`, `\"` escapes `"`
+- Pipes (`|`) in values require quoting
+- Enum values with spaces or commas need individual quotes
 
-Pipes (`|`) inside constraint values must be quoted to prevent interpretation as the type-constraint separator:
+## Schema Evolution Philosophy
 
-```yaml
-# Pipe in regex pattern - must quote the value
-format: 'string | pattern="a|b|c" default="x|y"'
+OpenChoreo schemas allow additional properties beyond what's defined (no `"additionalProperties": false`), enabling safe schema evolution:
 
-# Without quotes, this would be interpreted incorrectly
-# WRONG: pattern: string | pattern=a|b|c
+- **Development**: Add fields to Component before updating ComponentType schema
+- **Promotion**: Add new `envOverrides` in target environment before promoting Release with updated schema
```

Comment:
```suggestion
- **Promotion**: Add new `envOverrides` in target environment before promoting ComponentRelease with updated schema
```
Shall we use the term `ComponentRelease` instead of just `Release`, which may lead to confusion with the rendered Release?